numba/unicode.py

```diff
@@ -541,6 +541,73 @@ def find_impl(a, b):
         return find_impl
 
 
+@overload_method(types.UnicodeType, 'rfind')
+def unicode_rfind(s, substr, start=None, end=None):
+    """Implements str.rfind()"""
+    def unicode_rfind_check_type(ty, name):
+        """Check object belongs to one of specific types
+        ty: type
+            Type of the object
+        name: str
+            Name of the object
+        """
+        thety = ty
+        # if the type is omitted, the concrete type is the value
+        if isinstance(ty, types.Omitted):
+            thety = ty.value
+        # if the type is optional, the concrete type is the captured type
+        elif isinstance(ty, types.Optional):
+            thety = ty.type
+
+        accepted = (types.Integer, int, types.NoneType)
```

Comment:
I don't think that checking for a literal `int` is necessary in this case, unless you have a test-case to show otherwise.